src/pages/notebook.tsx

```diff
@@ -19,57 +21,104 @@ export const notebookPlugin: JupyterFrontEndPlugin<void> = {
     const contents = app.serviceManager.contents;
 
     const params = new URLSearchParams(window.location.search);
-    const notebookId = params.get('notebook');
+    let notebookId = params.get('notebook');
 
-    if (notebookId) {
-      // TODO replace with API call (and iteration over cells to set `editable: false`)
-      const content = {
-        cells: [
-          {
-            cell_type: 'code',
-            execution_count: null,
-            id: '55eb9a2d-401d-4abd-b0eb-373ded5b408d',
-            metadata: {
-              // This makes cell non-editable
-              editable: false
-            },
-            outputs: [],
-            source: [`# This is notebook '${notebookId}'`]
-          }
-        ],
-        metadata: {
-          kernelspec: {
-            display_name: 'Python 3 (ipykernel)',
-            language: 'python',
-            name: 'python3'
-          },
-          language_info: {
-            codemirror_mode: {
-              name: 'ipython',
-              version: 3
-            },
-            file_extension: '.py',
-            mimetype: 'text/x-python',
-            name: 'python',
-            nbconvert_exporter: 'python',
-            pygments_lexer: 'ipython3'
-          }
-        },
-        nbformat: 4,
-        nbformat_minor: 5
-      };
-      contents
-        .save('Untitled.ipynb', {
+    if (notebookId && notebookId.endsWith('.ipynb')) {
+      notebookId = notebookId.slice(0, -6);
+    }
+
+    /**
+     * Load a shared notebook from the CKHub API
+     */
+    const loadSharedNotebook = async (id: string): Promise<void> => {
+      try {
+        console.log(`Loading shared notebook with ID: ${id}`);
+
+        const apiUrl = 'http://localhost:8080/api/v1';
+        const sharingService = new SharingService(apiUrl);
+
+        console.log(`API URL: ${apiUrl}`);
+        console.log('Retrieving notebook from API...');
+
+        const notebookResponse = await sharingService.retrieve(id);
+        console.log('API Response received:', notebookResponse); // debug
+
+        const content: INotebookContent = notebookResponse.content;
+
+        // We make all cells read-only by setting editable: false
+        // by iterating over each cell in the notebook content.
+        if (content.cells) {
+          content.cells.forEach(cell => {
+            if (!cell.metadata) {
+              cell.metadata = {};
+            }
+            cell.metadata.editable = false;
+          });
+        }
+
+        if (!content.metadata) {
+          content.metadata = {};
+        }
+        content.metadata.isSharedNotebook = true;
+        content.metadata.sharedId = notebookResponse.id;
+        content.metadata.readableId = notebookResponse.readable_id;
+        content.metadata.domainId = notebookResponse.domain_id;
```

Comment:
I think you can use object destructuring here, i.e.

```typescript
content.metadata = {
  ...content.metadata,
  isSharedNotebook: true,
  <rest of things here>
}

same with up above, where you modify `cell.metadata.editable`.

Reply:
Ah, right, I keep forgetting about what the right syntax for destructuring is 😅

Reply:
Addressed in 61a6030dd3575be94687159fda6d18294304ef23, thanks!